doc.go

```diff
@@ -0,0 +1,247 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+/*
+package ttrpc defines and implements a low level simple transfer protocol
+optimized for low latency and reliable connections between processes on the same
+host. The protocol uses simple framing for sending requests, responses, and data
+using multiple streams.
+
+# Protocol
+
+The ttrpc protocol uses simple frames for communicating messages between a
+client and a server. The client represents the process which initiated the
+underlying connection and the server is the process which accepted the
+connection. The protocol is currently defined as asymmetrical, with clients
+sending requests and servers sending responses. Both clients and servers are
+able to send stream data. The roles are also used in determining the stream
+identifiers, with client initiated streams using odd number identifiers and
+server initiated using even number. The protocol may be extended in the future
+to support server initiated streams, that is not supported in the latest
+version.
+
+## Message Frame
+
+Each Message Frame consists of a 10-byte message header followed
+by message data. The data length and stream ID are both big-endian
+4-byte unsigned integers. The message type is an unsigned 1-byte
+integer. The flags are also an unsigned 1-byte integer and
+use is defined by the message type.
+
+    +---------------------------------------------------------------+
```

Comment:
just curiosity, did you draw this manually? :joy:

Reply:
half yes, used format from https://datatracker.ietf.org/doc/html/rfc7540#section-6.1